test/e2e/performanceprofile/functests/2_performance_update/tuned_deferred.go

```diff
@@ -0,0 +1,320 @@
+				case "deferred":
+					out, err := verifyDeferredApplication(tuned, workerCNFNodes, tc.Kernel_shmmni, name)
+					Expect(err).To(BeNil())
+					Expect(out).To(BeTrue())
+					rebootNode(poolName)
+					verifyImmediateApplication(tuned, workerCNFNodes, tc.Kernel_shmmni)
+				}
+			},
+			Entry("[test_id:78115] Verify deferred Always with first-time profile change", TunedProfileConfig{
+				DeferMode:         "always",
+				ProfileChangeType: "first-time",
+				ExpectedBehavior:  "deferred",
+				Kernel_shmmni:     "8192",
+			}),
+			Entry("[test_id:78116] Verify deferred Always with in-place profile update", TunedProfileConfig{
+				DeferMode:         "always",
+				ProfileChangeType: "in-place",
+				ExpectedBehavior:  "deferred",
+				Kernel_shmmni:     "4096",
+			}),
+			Entry("[test_id:78117] Verify deferred Update with first-time profile change", TunedProfileConfig{
+				DeferMode:         "update",
+				ProfileChangeType: "first-time",
+				ExpectedBehavior:  "immediate",
+				Kernel_shmmni:     "8192",
+			}),
+			Entry("[test_id:78118] Verify deferred Update with in-place profile update", TunedProfileConfig{
+				DeferMode:         "update",
+				ProfileChangeType: "in-place",
+				ExpectedBehavior:  "deferred",
+				Kernel_shmmni:     "4096",
+			}),
+			Entry("[test_id:78119] Verify deferred with No annotation with first-time profile change", TunedProfileConfig{
+				DeferMode:         "",
+				ProfileChangeType: "first-time",
+				ExpectedBehavior:  "immediate",
+				Kernel_shmmni:     "8192",
+			}),
+			Entry("[test_id:78120] Verify deferred Never mode with in-place profile update", TunedProfileConfig{
+				DeferMode:         "never",
+				ProfileChangeType: "in-place",
+				ExpectedBehavior:  "immediate",
+				Kernel_shmmni:     "4096",
+			}),
+		)
+	})
+
+	AfterAll(func() {
+		tuned = getTunedProfile(name)
+		if tuned != nil {
+			Expect(testclient.ControlPlaneClient.Delete(context.TODO(), tuned)).To(Succeed())
+		}
+	})
+
+})
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func getTunedProfile(tunedName string) *tunedv1.Tuned {
+	tunedList := &tunedv1.TunedList{}
+	var matchedTuned *tunedv1.Tuned
+
+	Eventually(func(g Gomega) {
+		g.Expect(testclient.DataPlaneClient.List(context.TODO(), tunedList, &client.ListOptions{
+			Namespace: components.NamespaceNodeTuningOperator,
+		})).To(Succeed())
+
+		g.Expect(len(tunedList.Items)).To(BeNumerically(">", 1))
+
+		for _, tuned := range tunedList.Items {
+			if tuned.Name == tunedName {
+				matchedTuned = &tuned
+				break
+			}
+		}
+
+	}).WithTimeout(time.Minute * 3).WithPolling(time.Second * 10).Should(Succeed())
+
+	if matchedTuned == nil {
+		return nil
+	}
+	return matchedTuned
+}
+
+func createTunedObject(deferMode string, kernel_shmmni string, np *hypershiftv1beta1.NodePool, name string) (*tunedv1.Tuned, error) {
+	tunedName := name
+	ns := components.NamespaceNodeTuningOperator
+	priority := uint64(19)
```

Comment:
why do we need these intermediate variables?

Reply:
@ffromani I would need to bring back the priority value, as it is failing our CI - [Link](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_cluster-node-tuning-operator/1230/pull-ci-openshift-cluster-node-tuning-operator-master-e2e-gcp-pao-updating-profile/1869647171192623104#1:build-log.txt%3A1153)